libpod/pod.go

```diff
@@ -138,46 +103,109 @@ func (p *Pod) Namespace() string {
 
 // ResourceLim returns the cpuset resource limits for the pod
 func (p *Pod) ResourceLim() *specs.LinuxResources {
-	resCopy := &specs.LinuxResources{}
-	if err := JSONDeepCopy(p.config.InfraContainer.ResourceLimits, resCopy); err != nil {
-		return nil
-	}
-	if resCopy != nil && resCopy.CPU != nil {
-		return resCopy
-	}
+	resCopy := &specs.Linux{}
 	empty := &specs.LinuxResources{
 		CPU: &specs.LinuxCPU{},
 	}
+	infra, err := p.runtime.GetContainer(p.state.InfraContainerID)
+	if err != nil {
+		return empty
+	}
+	conf, err := infra.specFromState()
+	if err != nil {
+		return empty
+	}
+	if err = JSONDeepCopy(conf.Linux, resCopy); err != nil {
```

Comment:
Should just deep-copy Resources. Check if Linux and Resources are nil and return nil if so, otherwise deep copy Resources